### DISCUSSION TITLE
Peripheral selection logic

### DISCUSSION THREAD

**Discussion by KenWillmott** [KenWillmott/HD6303EVB#11] [General]
The IOSEL* and BSEL* bus signals were intended to simplify peripheral select logic. In theory, two ways. By providing a select that is limited to a defined memory area, and by including the E strobe that is always needed to gate a peripheral IC.

However, I'm having second thoughts about this. Although it works in simple cases (I already connected one of them to the NVRAM address latch for example), I can see that in cases where the select must be further subdivided for multiple devices, it may not be appropriate. This is because of the propagation delays through the select logic, which may create a late write pulse if there are too many intervening logic stages. Typical 68xx systems cascade address and control signal logic FIRST, and the R/W pulse timing for a peripheral or memory IC, is provided by gating with E at the last possible stage. This allows all the address/control logic to settle before it's used. A write operation is terminated at the falling edge of E, if it is delayed, the hold time of the peripheral or memory device may be exceeded.

It's not obvious how to improve this. It's easy to remove E from the select circuit and expect expansion cards to do the gating, at the expense of usually one more IC. Another way, to leave the selects as they are, and expect expansion cards to do full decoding in the available range starting at 0xC000. Looking at a few different peripheral IC's, many need some adaptation to work with the 68xx bus at all, and the select line feature is not helpful for that. In fact, in some cases it seemed to make things more difficult.

So, it's an idea that worked well in the small "sandbox" design, but doesn't seem scalable to multiple expansion boards. Unfortunately, it has ramifications for the main board, as it sets the standard for timing.

Generally, I want to keep the one select signal per slot, since it can concentrate a lot of the decoding in one place on the main board, instead of being duplicated on every expansion board. I'm leaning towards utilizing the existing gated logic only on the main board itself, and generating additional selects for any expansion slots.

**Comment by TheEccentricGenius** [comment]
I think from reading your post, it would be good to keep the existing gated logic on the main board and use additional selects for expansions.
